### PR TITLE
Article added for Circular Linked list

### PR DESCRIPTION
**Pull request by madihamallick**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

(Write your answer here.)

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes : #74

### CODE REVIEW

File: docs/data-structures/Circular-Linked-list.mdx [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+    - **Average Case**: O (n)
+    - **Worst Case**: O (n)
+- Insertion
+    - **Average Case**: O (1)
+    - **Worst Case**: O (1)
+- Deletion
+    - **Average Case**: O (1)
+    - **Worst Case**: O (1)
+
+- Space Complexity: O (n)
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:451]: Keep all the values in backticks

File: docs/data-structures/Circular-Linked-list.mdx [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+1
+enter position of number 
+3
+```
+
+#### Sample Output
+
+```
+1 -> 3 -> 7 -> 4
+```
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:434]: The I/O sample of C++ and Python doesn't seem to match. Can you fix?

File: docs/data-structures/Circular-Linked-list.mdx [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+
+## Algorithm
+
+Here are the operations performed on a Circular Linked List:
+
+#### Traversing a Linked list: 
+
+- While the pointer containing address of the next node is not NULL
+- Fetch the data part of the node
+- Advance the pointer to the next node
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:36]: Put full-stops

File: docs/data-structures/Circular-Linked-list.mdx [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Inserting node at the beginning of the Linked list:
+
+- Allocate memory for the the new node and initialize its data part
+- Take a pointer PTR pointing to the start node of the list, and move it so that now it points to the last node
+- Add the new node between start pointer and PTR, and mark the new node as the start node
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:42]: Put full-stops

File: docs/data-structures/Circular-Linked-list.mdx [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Inserting node at the end of the Linked list:
+
+- Allocate memory for the the new node and initialize its data part
+- Take a pointer PTR pointing to the start node of the list, and move it so that now it points to the last node
+- Add the new node after the node pointed by PTR
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:48]: Put full-stops

File: docs/data-structures/Circular-Linked-list.mdx [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Inserting node at a given position of the Linked list:
+
+- Count the number of data items in the Circular Linked list
+- If the inserting position is 1, insert at the beginning
+- if the inserting position is at the end, insert at the end
+- if it is in the middle, link the newly created node with its left and right node
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:55]: Put full-stops

File: docs/data-structures/Circular-Linked-list.mdx [hunk 7]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Deleting node from the beginning of the Linked list:
+
+- Take a pointer PTR pointing to the start node of the list, and move it so that now it points to the last node
+- The link part of the last node is now made to point to the second node of the list
+- The memory occupied by the fisrt node is freed
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:61]: Put full-stops

File: docs/data-structures/Circular-Linked-list.mdx [hunk 8]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Deleting node from the end of the Linked list:
+
+- we move to last node by setting present node as ptr and previous one as preptr
+- when we reach the last node i.e. when last node is ptr and second last preptr
+- Now make the preptr's link part store the address of the first node, and free the space allocated for PTR
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:67]: Put full-stops

File: docs/data-structures/Circular-Linked-list.mdx [hunk 9]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Deleting node from a given position of the Linked list:
+
+- First we find the previous node of the node that is to be deleted
+- Change the address part of the previous node 
+- Free the memory occupied by the node
```
Comment (HarshCasper) [inline docs/data-structures/Circular-Linked-list.mdx:73]: Put full-stops

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/data-structures/Circular-Linked-list.mdx (hunk 2 above)
Comment (madihamallick) [reply docs/data-structures/Circular-Linked-list.mdx]: > The I/O sample of C++ and Python doesn't seem to match. Can you fix?

I could make then exactly the same, but I changed some parts. Now they're somewhat same.

**Comment by madihamallick** [comment]
@HarshCasper I have made the changes. Please review.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**